### PR TITLE
mysql dhcp pool backend

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Added support for mysql as dhcp pool backend (multiples active dhcp server running)

# Impacts
DHCP

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* Support for MySQL as dhcp pool backend and provide active/active dhcp support
image

### CODE REVIEW

File: go/dhcp/pool/mysql.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+// GetIssues Compare what we have in the cache with what we have in the pool
+func (dp *Mysql) GetIssues(macs []string) ([]string, map[uint64]string) {
+	t := dp.DHCPPool.NewTiming()
+	defer dp.DHCPPool.timeTrack(t, "GetIssues")
+	var inPoolNotInCache []string
+	var duplicateInPool map[uint64]string
+	duplicateInPool = make(map[uint64]string)
+
+	// TODO
```
Comment (julsemaan) [inline go/dhcp/pool/mysql.go]: should we do it ? A simple SQL query will likely give us that info

File: html/pfappserver/lib/pfappserver/Form/Config/Network.pm [hunk 2]
```diff
@@ -61,7 +61,18 @@ has_field 'algorithm' =>
...
   );
-has_field 'ip_reserved' =>
+has_field 'pool_backend' =>
+  (
+   type => 'Select',
+   label => 'DHCP Pool Backend Type',
+   required => 1,
+   options => [
+        { value => $pf::constants::dhcp::MEMORY_POOL, label => 'Memory Pool'},
+        { value => $pf::constants::dhcp::MYSQL_POOL, label => 'Mysql Pool'},
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Network.pm]: MySQL

File: html/pfappserver/root/static.alt/src/views/Configuration/_config/layer2Network.js [hunk 3]
```diff
@@ -31,6 +31,16 @@ export const dhcpList = [
   { value: '2', text: i18n.t('Oldest Released') }
 ]
 
+export const DHCPPoolTypes = [
+  { value: 'memory', text: i18n.t('Memory Pool') },
+  { value: 'mysql', text: i18n.t('Mysql Pool') }
```
Comment (julsemaan) [inline html/pfappserver/root/static.alt/src/views/Configuration/_config/layer2Network.js]: MySQL

File: html/pfappserver/root/static.alt/src/views/Configuration/_config/layer2Network.js [hunk 4]
```diff
@@ -118,6 +134,22 @@ export const view = (form = {}, meta = {}) => {
...
+          cols: [
+            {
+              namespace: 'pool_backend',
+              component: pfFormChosen,
+              attrs: {
+                collapseObject: true,
+                placeholder: i18n.t('Select a backend'),
+                trackBy: 'value',
+                label: 'text',
+                options: DHCPPoolTypes
```
Comment (julsemaan) [inline html/pfappserver/root/static.alt/src/views/Configuration/_config/layer2Network.js]: You should probably be able to get those from meta instead

File: html/pfappserver/root/static.alt/src/views/Configuration/_config/routedNetwork.js [hunk 5]
```diff
@@ -49,6 +49,16 @@ export const htmlNote = `<div class="alert alert-warning">
   ${i18n.t('Adding or modifying a network requires a restart of the pfdhcp and pfdns services for the changes to take place.')}
 </div>`
 
+export const DHCPPoolTypes = [
+  { value: 'memory', text: i18n.t('Memory Pool') },
+  { value: 'mysql', text: i18n.t('Mysql Pool') }
```
Comment (julsemaan) [inline html/pfappserver/root/static.alt/src/views/Configuration/_config/routedNetwork.js]: MySQL

File: html/pfappserver/root/static.alt/src/views/Configuration/_config/routedNetwork.js [hunk 6]
```diff
@@ -231,6 +248,22 @@ export const view = (form = {}, meta = {}) => {
...
+          cols: [
+            {
+              namespace: 'pool_backend',
+              component: pfFormChosen,
+              attrs: {
+                collapseObject: true,
+                placeholder: i18n.t('Select a backend'),
+                trackBy: 'value',
+                label: 'text',
+                options: DHCPPoolTypes
```
Comment (julsemaan) [inline html/pfappserver/root/static.alt/src/views/Configuration/_config/routedNetwork.js]: from meta here should also be possible

**Comment by fdurand** [comment]
All as been fixed, ready to review or merge

**Pull request merged**